miasm/arch/arm/sem.py

```diff
@@ -974,31 +975,45 @@ def blx(ir, instr, a):
 # todo
 def vmov(ir, instr, a, b, c=None):
     e = []
+    if c is None:
+        if a not in spregs_expr and b not in spregs_expr:
+            raise NotImplementedError('Not implemented')
+        e.append(ExprAssign(a, b))
+    else:
+        if a in dpregs_expr:
+            # two gp to one dp
+            # a[32:] = b
+            # a[:32] = c
+            e.append(ExprAssign(a, (c.zeroExtend(a.size) << ExprInt(32, 64) | b.zeroExtend(a.size))))
+        elif a in regs_expr:
+            # one dp to two gp
+            # a = c[32:]
+            # b = c[:32]
+            e.append(ExprAssign(a, (c & ExprInt(0xffffffff, 64))[0:32]))
```

Comment:
Hey, you are close to a good semantic.
Just a remark here, as you are doing a [0:32] (which will extract the first 32 bits) you don't need to mask it before with the  `& 0xFFFFFFFF`